src/Protocol/Protocol_1_9.cpp

```diff
@@ -1233,6 +1233,28 @@ void cProtocol_1_9_0::SendResetTitle(void)
 
 
 
+void cProtocol_1_9_0::SendResourcePack(const AString & a_ResourcePackUrl)
+{
+	cPacketizer Pkt(*this, pktResourcePack);
+
+	cSha1Checksum Checksum;
+	Checksum.Update(reinterpret_cast<const Byte *>(a_ResourcePackUrl.c_str()), a_ResourcePackUrl.size());
+	Byte Digest[20];
+	Checksum.Finalize(Digest);
+	AString Sha1Output;
+	for (size_t i = 0; i < ARRAYCOUNT(Digest); i++)
+	{
+		AppendPrintf(Sha1Output, "", Digest[i]);
+	}
```

Comment:
There's a `cSha1Checksum::DigestToJava` function, if that is not usable here, perhaps add a new one there, `DigestToHex`.